### ISSUE TITLE
Detection is configured differently on the browser and server

### ISSUE THREAD

**Issue by kshahani** [isaachinman/next-i18next#826]
## Describe the bug

I have the following detection settings:

```ts
const detection = {
  // Prioritize query string to allow forcing the user language
  order: ['querystring', 'cookie', 'header'],
  lookupQuerystring: 'lang',
};
```

I needed to initialize the next-i18next instance like this:

```ts
export const nextI18next = new NextI18Next({
  // other properties
  detection, // initializes browser i18n instance detection config
  ...detection, // initializes server i18n instance detection config
});
```

If I didn't have the latter spread, I'd have a flash of content since the server would rely on the cookie for the language (say in English), and the browser would then swap the language to French by reading the query parameter.

## Occurs in next-i18next version
6.0.2

## Steps to reproduce
Provided above

## Expected behaviour
Both server and browser detection are configured from the detection object.

**Comment by isaachinman** [comment]
@kshahani I cannot reproduce that. If I add this detection setup to the simple example (without any spread), it works as expected. Please provide a reproducible example.

**Comment by kshahani** [comment]
@isaachinman Thanks for reply. I'll provide an example repo today.

**Comment by nouh** [comment]
some detection options are different on browser and server, but it seems  right now they use the same detection config?

**Comment by isaachinman** [comment]
Hi there, I have just released `next-i18next@8.0.0`, which uses the internationalised routing built into NextJs v10+. This new version of `next-i18next` is quite different, thus making this PR no longer relevant. Thanks!
